SDWebImage/Private/SDDisplayLink.m

```diff
@@ -219,17 +241,17 @@ - (void)stop {
 }
 
 - (void)displayLinkDidRefresh:(id)displayLink {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+    [_target performSelector:_selector withObject:self];
```

Comment:
The first callback (in `SDAnimatedImagePlayer`) will get an temp status about `duration`, because at this time, the previous `targetTimestamp` or `timestamp` is not set. We have some fallback logic

Then the second callback the duration will be accurate enough